src/sage/numerical/pivot_rules_for_simplex_method.py

```diff
@@ -0,0 +1,776 @@
+            sage: D = P.initial_dictionary()
+            sage: pivot = SimplexMethodPivot("steepest_edge") # indirect test
+            sage: pivot(D, False, 1) # let's use the 1 norm.
+            sage: D.entering()
+            x1
+            sage: D.leaving()
+            x4
+
+            Values of p are checked::
+
+            sage: D.update()
+            sage: pivot(D, False, 0)
+            Traceback (most recent call last):
+            ...
+            ValueError: p must be at least 1
+        """
+        if p < 1:
+            raise ValueError("p must be at least 1")
+        if current_dictionary.entering() is None:
+            thetas = []
+            for v in current_dictionary.basic_variables():
+                # We want to avoid p-root complications, so work with the pth power.
+                thetas.append((1+sum([i**p for i in current_dictionary.row_coefficients(v)])))
+            best_value = -1*current_dictionary.objective_coefficients()[0]/thetas[0]
+            index_to_select = 0
+            # This loop selects the index that corresponds to the steepest edge.
+            for i in range(1,len(current_dictionary.nonbasic_variables())):
+                decent_direction = -1*current_dictionary.objective_coefficients()[i]/thetas[i-1]
+                if decent_direction < best_value:
+                    best_value = decent_direction
+                    index_to_select = i
+            entering_variable = current_dictionary.nonbasic_variables()[index_to_select]
+            current_dictionary.enter(entering_variable)
+        if current_dictionary.leaving() is None:
+            possible = current_dictionary.possible_leaving()
+            if possible:
+                # We select leaving variables based on ratios.
+                leaving_variable = min(current_dictionary.ratios())[1]
+                current_dictionary.leave(leaving_variable)
+
+class DantzigsRule(AbstractSimplexMethodPivotRule):
+    r"""
+    Selects the entering variable to be the variable with the most negative reduced cost.
+
+    Originally proposed by Dantzig in the formulation of linear programming. Also know as greatest improvement rule.
+
+    See [TZ1993]_.
+
+    EXAMPLES::
+
+        sage: from sage.numerical.pivot_rules_for_simplex_method import SimplexMethodPivot
+        sage: A = ([1, 1], [3, 1])
+        sage: b = (1000, 1500)
+        sage: c = (10, 5)
+        sage: P = InteractiveLPProblemStandardForm(A, b, c)
+        sage: D = P.initial_dictionary()
+        sage: D.entering()
+
+        sage: D.leaving()
+
+        sage: pivot = SimplexMethodPivot("dantzig")
+        sage: pivot(D)
+        sage: D.entering()
+        x1
+        sage: D.leaving()
+        x4
+    """
+    def dictionary_pivot(current_dictionary):
+        r"""
+        Sets the entering and leaving variables of :class:`sage.numerical.interactive_simplex_method.LPDictionary` as defined by Dantzig's rule.
```

Comment:
The first line/sentence of the documentation of a method needs to be written in imperative mood: "Set the entering and ..."